### DISCUSSION TITLE
Figures Dimensions

### DISCUSSION THREAD

**Discussion by friosavila** [hugetim/nbstata#24] [General]
Hi Tim
I was wondering if there is anyway to modify how nbstata works regarding figures.
I know that currently, you can change the setup if you would like to obtain larger or smaller figures to be output in jupyter notebook.
However, would it be possible to use information from "xsize() and ysize()" to be used to change those settings in a more dynamic way?

Thank you!

**Comment by hugetim** [comment]
Have you already tried it to see if it works currently? If it doesn't, I'm
not sure what can be done about it. It may be a limitation of `pystata`,
which is what underlies `nbstata`.

**Comment by friosavila** [comment]
Hi Tim
I tried and it doesn’t work.
It generally produces a figure that is reshaped to fit the default
parameters (6x4 I think?)
I will try it again in a notebook using nbstata and pystata to see if it
works
Thank you!
Fernando

**Comment by friosavila** [comment]
Just as feedback
I just tried the "size" options.
when using nbstata, figures that are shown on the notebook are unaffected
when using "size" "ysize" options.
However, if I do the same using pystata (with magic commands %%stata),
figures do adjust as needed.
I can send you the notebooks if they are useful
Fernando

**Reply by hugetim** [reply]
Hmm, that's a bit surprising. I guess maybe that means there's a difference between the behavior of `%%stata` and `pystata.stata.run` (which is what nbstata uses under the hood). I am curious to see the notebooks if they're easy to send.

**Comment by friosavila** [comment]
SUre thing
Here is the minimal example

**Reply by hugetim** [reply]
Hmm, maybe you forgot to attach something?

**Reply by hugetim** [reply]
But nevermind. I've tried it out for myself now and see what you're saying. It's not a difference between `%%stata` and `pystata.run` as I guessed. Rather it's an [undocumented](pystata.config.set_graph_size) side effect of `pystata.config.set_graph_size`. If you set graph size to something other than "default", it stops the xsize/ysize options from impacting. But if you leave set_graph_size as ("default", "default"), xsize/ysize work (with both `%%stata` and `pystata.run`). As a result, I should probably change the `nbstata` treatment of graph_size, leaving the "default"s in place...

**Comment by hugetim** [comment]
Just in case, did you see that you can use the `%%set` magic to change the graph size within a notebook? https://hugetim.github.io/nbstata/user_guide.html#set-set

**Comment by friosavila** [comment]
Thank you!
I think the attachment got lost because I send it as a reply to github?
What Im ending up doing is quietly (and with output: false) produce the
graphs, export them into other formats (png or pdf)
and then add that figure into my document.
Have become a bit more proficient with QUarto-nbstata but few things are
still hard in regards to tables and referencing.
If I encounter anything else, will let you know
Fernando

**Comment by hugetim** [comment]
Ok, there is a workaround (not yet documented) that will make the `xsize` and `ysize` options work for a graph command in a Stata notebook:
https://github.com/hugetim/nbstata/issues/29#issuecomment-1703084824
